test/jdk/javax/swing/JToggleButton/bug4128979.java

```diff
@@ -22,63 +22,67 @@
-   @summary Tests that background changes correctly in WinLF for JToggleButton when pressed
-   @key headful
-   @run applet/manual=yesno bug4128979.html
-*/
+ * @bug 4128979
+ * @requires (os.family == "windows")
+ * @modules java.desktop/sun.awt
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @summary Tests that background changes correctly in WinLF for JToggleButton when pressed
+ * @run main/manual bug4128979
+ */
+
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.ComponentOrientation;
+import java.awt.Container;
+import java.awt.FlowLayout;
+import java.awt.Graphics;
 
 import javax.swing.BorderFactory;
 import javax.swing.BoxLayout;
 import javax.swing.Icon;
-import javax.swing.JApplet;
 import javax.swing.JFrame;
 import javax.swing.JLabel;
-import javax.swing.JOptionPane;
 import javax.swing.JPanel;
 import javax.swing.JToggleButton;
 import javax.swing.JToolBar;
 import javax.swing.UIManager;
-import javax.swing.UnsupportedLookAndFeelException;
-import javax.swing.WindowConstants;
-import java.awt.BorderLayout;
-import java.awt.Color;
-import java.awt.Component;
-import java.awt.ComponentOrientation;
-import java.awt.Container;
-import java.awt.FlowLayout;
-import java.awt.Graphics;
 
-public class bug4128979 extends JApplet {
+import jtreg.SkippedException;
+import sun.awt.OSInfo;
 
-    public static void main(String[] args) {
-        JApplet applet = new bug4128979();
-        applet.init();
-        applet.start();
+public class bug4128979 {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS = """
+            When the test starts, toggle buttons are visible in three rows
+            two of which are toolbars.
 
-        JFrame frame = new JFrame("Test window");
-        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
-        frame.setLayout(new BorderLayout());
-        frame.add(applet, BorderLayout.CENTER);
-        frame.setSize(600, 240);
-        frame.setVisible(true);
-    }
+            Press these buttons, their background color must change
+            to half tones between the button background colors and the ToggleButton
+            highlight color (it is shown in the square below).
 
-    public void init() {
-        try {
-            UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsClassicLookAndFeel");
-        } catch (UnsupportedLookAndFeelException e) {
-            JOptionPane.showMessageDialog(this,
-                    "This test requires Windows look and feel, so just press Pass\n as "+
-                    " this look and feel is unsupported on this platform.",
-                    "Unsupported LF", JOptionPane.ERROR_MESSAGE);
-            return;
-        } catch (Exception e) {
-            throw new RuntimeException("Couldn't set look and feel");
+            If the background color does not change correctly for at least one button,
+            the test fails. """;
```

Comment:
Getting a white space after `fails.`, can you check once.

Reply:
Not a problem though since it is part of instruction. Anyways removed extra space.